lib/java_buildpack/framework/app_dynamics_agent.rb

```diff
@@ -83,7 +84,7 @@ def host_name(java_opts, credentials)
 
       def node_name(java_opts, credentials)
         name = credentials['node-name'] || @configuration['default_node_name']
-        java_opts.add_system_property('appdynamics.agent.nodeName', "#{name}")
+        java_opts.add_system_property('appdynamics.agent.nodeName', @appdynamics_tier_name + ":#{name}")
```

Comment:
Hi,
We do understand that the node_name is configurable by different ways. The reason why we propose to change the node-name is because  the default behaviour of the buildpack of mapping the node_name to the instance_index would not help customers who completely rely on the default behaviour of the BP OR are not that accustomed to CF CLI OR who are first time users of AppDynamics and after instrumenting their app see the node_name having values as 0,1,2 etc... replicated across all the tiers of their application.